docs/tutorials/my-first-custom-action.rst

```diff
@@ -0,0 +1,256 @@
+        spoiler: alert
+
+Now you start the pod by running the following command:
+
+.. code-block:: bash
+
+    $ kubectl apply -f nginx-pod.yaml
+    pod/nginx created
+
+but for some reason it it not starting (note it’s “Pending” Status):
+
+.. code-block:: bash
+
+    $ kubectl get pods
+
+    NAME                                                     READY   STATUS    RESTARTS   AGE
+
+    nginx                                                    0/1     Pending   0          5h19m
+
+You wait a few minutes, but is remains the same.
+
+Then you say, ok, let’s look at the event log:
+
+.. code-block:: bash
+
+    kubectl get event --field-selector involvedObject.name=nginx
+    LAST SEEN   TYPE      REASON             OBJECT      MESSAGE
+    64s         Warning   FailedScheduling   pod/nginx   0/1 nodes are available: 1 node(s) didn't match Pod's node affinity/selector.
+
+Aha! “1 node(s) didn't match Pod's node affinity/selector.”! ALRIGHT!!!
+
+
+
+Wait, what does it mean? 😖  (Hint: Check the YAML config for the spoiler)
+
+
+
+After searching online for some time, you find out that the YAML file you copied had a “nodeSelector” with the key-value “spoiler: alert”, which means that it can only be scheduled on nodes (machines) that have this configuration 🤦‍♂️.
+
+From the docs:
+
+.. pull-quote::
+    nodeSelector is the simplest recommended form of node selection constraint. nodeSelector is a field of PodSpec. It specifies a map of key-value pairs. For the pod to be eligible to run on a node, **the node must have each of the indicated key-value pairs as labels** (it can have additional labels as well). The most common usage is one key-value pair.
+
+So you comment out those lines, run kubectl apply again, and all is well.
+
+.. code-block:: yaml
+
+    apiVersion: v1
+    kind: Pod
+    metadata:
+      name: nginx
+      labels:
+        env: test
+    spec:
+      containers:
+      - name: nginx
+        image: nginx
+        imagePullPolicy: IfNotPresent
+    #  nodeSelector:
+    #    spoiler: alert
+
+
+Automating the detection with a Robusta Playbook
+----
+What we need to do?
+----
+
+.. note::
+    Make sure to clean up the pod from the last section by running ``kubectl delete pod nginx``
+
+A playbook consists of two things:
+
+- Trigger - We’re gonna use a built in trigger
+- Action - We’re gonna write our own action!
+
+
+Finding the correct trigger
+----
+So what is the correct trigger for the job?
+We can think of two triggers that may fit:
+
+- Creation of a new pod (because we create a new pod, ‘ngnix’)
+- A Kubernetes Event is fired (because we ran kubectl get event to find out the scheduling error)
+
+Let’s look at the Trigger section about :ref:`Kubernetes (API Server)`, and try to find out triggers for both.
+Go ahead and try to find them!
+
+Okay! We find ``on_pod_create`` and ``on_event_create``
+
+We’ll use ``on_event_create``, because in this case ``on_pod_create`` is not even called because the pod is not created.
```

Comment:
Are you sure about this? I would expect `on_pod_create` to be called because the Kubernetes pod resource is created... It even shows up in `kubectl get pods`.

The pod's status is pending and the pod was never scheduled, but the pod object itself was still created.

Reply:
When I tried it, it did not work. Maybe we can try together If you'd like. I don't want to change the rest of the tutorial, so maybe we can choose on_event_create for other reason, for example because it contains the information we need. What do you think?

Reply:
Sounds good to me. Happy to look at this together at some point.

For now, you can just write something like:

_We'll use `on_event_create` in this tutorial because it will be easier to identify scheduling issues by looking at the event._